os/se/ameba/security_ameba_wrapper.c

```diff
@@ -0,0 +1,1755 @@
+/**************************************************************************************************
+*
+* Copyright 2020 Samsung Electronics All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*
+***************************************************************************************************
+***************************************************************************************************
+*
+*   Copyright (C) 2020 Gregory Nutt. All rights reserved.
+*   Author: Gregory Nutt <gnutt@nuttx.org>
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions
+* are met:
+*
+* 1. Redistributions of source code must retain the above copyright
+*    notice, this list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in
+*    the documentation and/or other materials provided with the
+*    distribution.
+* 3. Neither the name NuttX nor the names of its contributors may be
+*    used to endorse or promote products derived from this software
+*    without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+* LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+* ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+***************************************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/seclink_drv.h>
+#include <tinyara/security_hal.h>
+
+#include <mbedtls/rsa.h>
```

Comment:
SE layer must not refer mbedtls which is in application layer. whereas SE layer is in kernel layer. so please remove all mbedtls functions in SE layer.

Reply:
RTL8721CSM are using software method to replace the Hardware SE function (ISP). In RTL8721CSM did not separate user/kernel build,  those it is directly using MbedTLS function. In future SE layer function of RTL8721CSM will be putting in TrustZone.

Reply:
Kernel layer must not call functions in application layer. I couldn't accept the PR if ameba wrapper called mbedtls. it breakes rule of TizenRT

Reply:
Sorry about that, we have modify the Se layer for RTL8721CSM, it will no longer calling mbedtls function. Please have a look on the latest commit